CHANGELOG.md

```diff
@@ -1,5 +1,9 @@
 ## 3.6.0 (YYYY-MM-DD)
 
+### Breaking Changes
+
+* [ObjectServer] Read-only is not supported by synced Realm. "SyncConfigration.readOnly()" will always throw `UnsupportedOperationException`.
```

Comment:
I don't know how Cocoa detect read only files, but they definitely have the same use case where people put Realm files on read-only storage. 

The second conversation you reference describes the motivation for treating missing tables as being present but just empty. I don't think we want these semantics in Java since it would complicate our query class quite a bit, and people can just create an explicit Schema to avoid the issue. Most likely this will require two different READ_ONLY modes in Object Store, which is a bit unfortunate.